src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/DeserializationHelpers.cs

```diff
@@ -580,6 +583,22 @@ private static object ConvertPrimitiveValue(string value)
             return deserializer.ReadInline(enumValue, propertyType, readContext);
         }
 
+        /// <summary>
+        /// Converts any escaped characters in the input string.
+        /// </summary>
+        /// <param name="inputString">The input string containing the text to convert.</param>
+        /// <returns>String with any escaped characters converted to their unescaped form.</returns>
+        private static string UnescapeString(string inputString)
+        {
+            if (string.IsNullOrWhiteSpace(inputString))
+            {
+                return inputString;
+            }
+
+            // Make use `Regex`'s `Unescape` method to pick out any escaped special characters and unescape them
+            return Regex.Unescape(inputString);
```

Comment:
>Regex [](start = 19, length = 5)

I remember that for some reason (security or performance), we try to remove "Regex" usage in the source code.
Do we have other way to overcome it?